public/app/plugins/panel/alertlist/unified-alerting/UngroupedView.tsx

```diff
@@ -1,69 +1,108 @@
+import { fromCombinedRule, stringifyIdentifier } from 'app/features/alerting/unified/utils/rule-id';
+import {
+  alertStateToReadable,
+  alertStateToState,
+  getFirstActiveAt,
+  isAlertingRule,
+} from 'app/features/alerting/unified/utils/rules';
+import { createUrl } from 'app/features/alerting/unified/utils/url';
 import { PromAlertingRuleState } from 'app/types/unified-alerting-dto';
 
+import { AlertingRule, CombinedRuleWithLocation } from '../../../../types/unified-alerting';
 import { AlertInstances } from '../AlertInstances';
 import { getStyles } from '../UnifiedAlertList';
 import { UnifiedAlertListOptions } from '../types';
 
 type UngroupedModeProps = {
-  rules: PromRuleWithLocation[];
+  rules: CombinedRuleWithLocation[];
   options: UnifiedAlertListOptions;
 };
 
 const UngroupedModeView: FC<UngroupedModeProps> = ({ rules, options }) => {
   const styles = useStyles2(getStyles);
   const stateStyle = useStyles2(getStateTagStyles);
+  const { href: returnTo } = useLocation();
 
   const rulesToDisplay = rules.length <= options.maxItems ? rules : rules.slice(0, options.maxItems);
 
   return (
     <>
       <ol className={styles.alertRuleList}>
         {rulesToDisplay.map((ruleWithLocation, index) => {
-          const { rule, namespaceName, groupName } = ruleWithLocation;
-          const firstActiveAt = getFirstActiveAt(rule);
-          return (
-            <li className={styles.alertRuleItem} key={`alert-${namespaceName}-${groupName}-${rule.name}-${index}`}>
-              <div className={stateStyle.icon}>
-                <Icon
-                  name={alertDef.getStateDisplayModel(rule.state).iconClass}
-                  className={stateStyle[alertStateToState(rule.state)]}
-                  size={'lg'}
-                />
-              </div>
-              <div>
-                <div className={styles.instanceDetails}>
-                  <div className={styles.alertName} title={rule.name}>
-                    {rule.name}
-                  </div>
-                  <div className={styles.alertDuration}>
-                    <span className={stateStyle[alertStateToState(rule.state)]}>
-                      {alertStateToReadable(rule.state)}
-                    </span>{' '}
-                    {firstActiveAt && rule.state !== PromAlertingRuleState.Inactive && (
-                      <>
-                        for{' '}
-                        <span>
-                          {intervalToAbbreviatedDurationString({
-                            start: firstActiveAt,
-                            end: Date.now(),
-                          })}
-                        </span>
-                      </>
-                    )}
+          const { namespaceName, groupName, dataSourceName } = ruleWithLocation;
+          const alertingRule: AlertingRule | undefined = isAlertingRule(ruleWithLocation.promRule)
+            ? ruleWithLocation.promRule
+            : undefined;
```

Comment:
Should we return null here if there is no `alertingRule`? this way we don't need to bother with `alertingRule` being undefined in the next lines